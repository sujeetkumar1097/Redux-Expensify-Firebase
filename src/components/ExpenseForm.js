import React from 'react'
import moment from 'moment'
import {SingleDatePicker} from 'react-dates'

export default class ExpenseForm extends React.Component {
    // state = {
    //     description: '',
    //     note: '',
    //     amount: '',
    //     createdAt: moment(),
    //     calenderFocused: false,
    //     error: ''
    // }
    constructor(props) {
        super(props)
        this.state = {
            description: props.expense ?  props.expense.description : '',
            note: props.expense ? props.expense.note : '',
            amount: props.expense ? (props.expense.amount /100).toString() : '',
            createdAt: props.expense ? moment(props.expense.createdAt) : moment(),
            calenderFocused: false,
            error: ''
        }
    }
    handleDescription = (e) => {
        const description = e.target.value  
        this.setState(() => ({description}))
    }
    handleAmount = (e) => {
        const amount = e.target.value
        if(!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
            this.setState(() => ({amount}))
        }
    }
    handleNote = (e) => {
        // e.persist();
        // this.setState(() => ({note: e.target.value}))
        const note = e.target.value
        this.setState(() => ({note}))
    }
    onDateChange = (date) => {
        if(date) {
            this.setState(() => ({createdAt: date}))
        }
    }
    onFocusChange= ({focused}) => {
        this.setState(() => ({calenderFocused: focused}))
    }
    handleSubmit = (e) => {
        e.preventDefault()

        if(!this.state.description || !this.state.amount) {
            this.setState(() => ({error: 'Please provide Description and Amount'}))
        } else {
            this.setState(() => ({error: ''}))
            this.props.onSubmit({
                description: this.state.description,
                amount: parseFloat(this.state.amount, 10)*100,
                createdAt: this.state.createdAt.valueOf(),
                note: this.state.note
            })
        }
    }

    render() {
        return (
                <form className="form" onSubmit={this.handleSubmit}>
                    {this.state.error && <p className="form__error">{this.state.error}</p>}
                    <input className="text-input" type='text' placeholder='Desc.' 
                            value={this.state.description}
                            onChange={this.handleDescription}/>
                    <input className="text-input" type='text' placeholder='Amt.' 
                            value={this.state.amount}
                            onChange={this.handleAmount}/>
                    <textarea className="textarea" placeholder='Add Note' 
                            value={this.state.note}
                            onChange={this.handleNote}/>
                    <SingleDatePicker
                        date = {this.state.createdAt}
                        onDateChange={this.onDateChange}
                        focused = {this.state.calenderFocused}
                        onFocusChange={this.onFocusChange}
                        numberOfMonths={1}
                        isOutsideRange={() => false}

                    />
                    <div>
                        <button className="button">ADD EXPENSE</button>
                    </div>
                </form>
        )
    }
}