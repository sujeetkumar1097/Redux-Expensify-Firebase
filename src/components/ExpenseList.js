import React from 'react'
import {connect} from 'react-redux'
import ExpenseListItem from './ExpenseListItems'
import ExpenseFilter from '../selectors/ExpenseFilter'

export const ExpenseList = (props) => (
    <div className="content-container">
        <div className="list-header">
            <div className="show-for-mobile">Expenses</div>
            <div className="show-for-desktop">Expense</div>
            <div className="show-for-desktop">Amount</div>
        </div>
        {/* {props.filters.text}
        {props.expenses.length} */}
        <div className="list-body">
            {
                props.expenses.length === 0 ? (
                    <div className="list-item list-item--message">
                        <span>No Expense Added Yet</span>
                    </div>
                ) : (
                    props.expenses.map(expense => 
                            (<ExpenseListItem key={expense.id} {...expense}/>)
                        )
                    )
            }
        </div>
    </div>
)

const mapStateToProps = (state) => {
    return {
        expenses: ExpenseFilter(state.expenses,state.filters),
        filters: state.filters
    }
}

export default connect(mapStateToProps)(ExpenseList)