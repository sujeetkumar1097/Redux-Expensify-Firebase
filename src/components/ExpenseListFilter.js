import React from 'react'
import {connect} from 'react-redux'
import {setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate} from '../actions/Filters'
import {DateRangePicker} from 'react-dates'
import moment from 'moment'

export class ExpenseListFilter extends React.Component {
    state = {
        calenderFocused: null
    }
    onDateChange = ({startDate, endDate}) => {
        this.props.setStartDate(startDate)
        this.props.setEndDate(endDate)
    } 
    onFocusChange = (calenderFocused) => {
        this.setState(() => ({calenderFocused}))
    }
    onTextChange =(e) => {
        this.props.setTextFilter(e.target.value)
    }
    onSelectChange = (e) => {
        if(e.target.value === 'date') {
            this.props.sortByDate()
        } else if(e.target.value === 'amount') {
            this.props.sortByAmount()
        }
    }
    render() {
        return (
            <div className="content-container">
                <div className="input-group">
                    <div className="input-group__item">
                        <input className="text-input" type='text' placeholder="Search Item..." value={this.props.filters.text} onChange={this.onTextChange}/>
                    </div>
                    <div className="input-group__item">
                        <select 
                            className="select"
                            value= {this.props.filters.sortBy}
                            onChange={this.onSelectChange}
                        >
                            <option value='date'>Date</option>
                            <option value='amount'>Amount</option>
                        </select>
                    </div>
                    <div className="input-group__item">
                        <DateRangePicker
                            startDate={this.props.filters.startDate}
                            endDate={this.props.filters.endDate}
                            onDatesChange={this.onDateChange}
                            focusedInput={this.state.calenderFocused}
                            onFocusChange={this.onFocusChange}
                            showClearDates={true}
                            numberOfMonths={1}
                            isOutsideRange={() => false}
                        />
                    </div>
                </div>
            </div>
        )
    }

}
const mapStateToProps = (state) => ({filters:state.filters})
const mapDispatchToProps = (dispatch) => ({
    setStartDate: (startDate) => dispatch(setStartDate(startDate)),
    setEndDate: (endDate) => dispatch(setEndDate(endDate)),
    setTextFilter: (text) => dispatch(setTextFilter(text)),
    sortByAmount: () => dispatch(sortByAmount()),
    sortByDate: () => dispatch(sortByDate()),
})

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilter)