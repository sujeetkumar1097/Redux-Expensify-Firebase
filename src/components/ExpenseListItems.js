import React from 'react';
import {NavLink} from 'react-router-dom'
import moment from 'moment'
import numeral from 'numeral'
 
const ExpenseListItems = ({ id, description, amount, createdAt}) => {
    //console.log("this is expense: obj ==>>",id);
    return (
            <NavLink className="list-item" to={`/edit/${id}`}>
                <div>
                    <h3 className="list-item__title">{description}</h3>
                    <span className="list-item__sub-title">{moment(createdAt).format('MMMM Do, YYYY')}</span>
                </div>
                <h3 className="list-item__amount">{numeral(amount/100).format('$0,0.00')}</h3>
            </NavLink>
    )
}
export default ExpenseListItems

