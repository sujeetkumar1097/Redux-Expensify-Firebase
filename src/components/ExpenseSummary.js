import React from 'react'
import {connect} from 'react-redux'
import {Link} from 'react-router-dom'
import ExpenseFilter from '../selectors/ExpenseFilter'
import ExpenseTotal from '../selectors/ExpenseTotal'
import numeral from 'numeral'

export const ExpenseSummary = ({expenseCount, expenseTotal}) => {
    const expenseWord = expenseCount === 1 ? 'Expense' : 'Expenses'
    const formattedExpensesTotal = numeral(expenseTotal/100).format('$0,0.00')
    
    return (
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">
                    Viewing <span>{expenseCount}</span> {expenseWord} totalling <span>{formattedExpensesTotal}</span>
                </h1>
                <div className="page-header__action">
                    <Link className="button" to="/expense">Add Expense</Link>
                </div>
            </div>
        </div>
    )
}

const mapStateToProps = (state) => {
    const visibleExpenses = ExpenseFilter(state.expenses, state.filters)
    return {
        expenseCount: visibleExpenses.length,
        expenseTotal: ExpenseTotal(visibleExpenses)
    }
}

export default connect(mapStateToProps)(ExpenseSummary)