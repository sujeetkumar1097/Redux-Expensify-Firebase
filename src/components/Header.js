import React from 'react';
import {NavLink, Link} from 'react-router-dom'
import {connect} from 'react-redux'
import {startLogOut} from '../actions/Authentication'


export const Header = ({startLogOut}) => (
    <div className="header">
        {/* <NavLink to='/dashboard' activeClassName='is-active'>Dashboard</NavLink>
        <NavLink to='/create' activeClassName='is-active'>Create Expense</NavLink> */}
        {/* <NavLink to='/edit'>Edit Expense</NavLink> */}
        {/* <NavLink to='/help' activeClassName=''>Help</NavLink> */}
        <div className="content-container">
            <div className="header__content">
                <Link className="header__title" to='/dashboard'>
                    <h1>Expensify</h1>
                </Link>
                <button className="button button--link" onClick={startLogOut}>LOGOUT</button>
            </div>
        </div>
        
    </div>
)

const mapDispatcherToProp = (dispatch) => ({
    startLogOut: () => dispatch(startLogOut())
})

export default connect(undefined, mapDispatcherToProp)(Header)