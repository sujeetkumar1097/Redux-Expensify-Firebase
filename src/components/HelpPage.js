import React from 'react';

const HelpPage = () => (
    <h1>Inside HelpPage</h1>
)

export default HelpPage