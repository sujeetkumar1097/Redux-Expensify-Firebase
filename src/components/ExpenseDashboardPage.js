import React from 'react';
import ExpenseList from './ExpenseList'
import ExpenseListFilter from './ExpenseListFilter'
import ExpenseSummary from './ExpenseSummary'

const ExpenseDashboardPage = () => (
    <div>
        <ExpenseSummary/>
        <ExpenseListFilter/>
       <ExpenseList/>
    </div>
    
)

export default ExpenseDashboardPage