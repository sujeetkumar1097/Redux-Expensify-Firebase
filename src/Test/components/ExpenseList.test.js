import React from 'react'
import {shallow} from 'enzyme'
import  {ExpenseList} from '../../components/ExpenseList'
import expenses from '../Fixtures/Expenses'

test('Expense List with Expenses', () => {
    const wrapper = shallow(<ExpenseList expenses={expenses}/>)
    expect(wrapper).toMatchSnapshot()
})

test('Expense List with Expenses', () => {
    const wrapper = shallow(<ExpenseList expenses={[]}/>)
    expect(wrapper).toMatchSnapshot()
})