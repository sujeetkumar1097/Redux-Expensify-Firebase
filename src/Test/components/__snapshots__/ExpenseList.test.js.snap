// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expense List with Expenses 1`] = `
<div
  className="content-container"
>
  <div
    className="list-header"
  >
    <div
      className="show-for-mobile"
    >
      Expenses
    </div>
    <div
      className="show-for-desktop"
    >
      Expense
    </div>
    <div
      className="show-for-desktop"
    >
      Amount
    </div>
  </div>
  <div
    className="list-body"
  >
    <ExpenseListItems
      amount={195}
      createdAt={0}
      description="Gum"
      id="1"
      key="1"
      note=""
    />
    <ExpenseListItems
      amount={109500}
      createdAt={-345600000}
      description="Rent"
      id="2"
      key="2"
      note=""
    />
    <ExpenseListItems
      amount={4500}
      createdAt={345600000}
      description="Credit Card"
      id="3"
      key="3"
      note=""
    />
  </div>
</div>
`;

exports[`Expense List with Expenses 2`] = `
<div
  className="content-container"
>
  <div
    className="list-header"
  >
    <div
      className="show-for-mobile"
    >
      Expenses
    </div>
    <div
      className="show-for-desktop"
    >
      Expense
    </div>
    <div
      className="show-for-desktop"
    >
      Amount
    </div>
  </div>
  <div
    className="list-body"
  >
    <div
      className="list-item list-item--message"
    >
      <span>
        No Expense Added Yet
      </span>
    </div>
  </div>
</div>
`;
